tests/detect_machine.sh

```diff
@@ -51,16 +51,14 @@ case $(hostname -f) in
   login0[1-2].expanse.sdsc.edu) MACHINE_ID=expanse ;; ### expanse1-2
 
   discover3[1-5].prv.cube) MACHINE_ID=discover ;; ### discover31-35
-  *) MACHINE_ID=UNKNOWN ;;  # Unknown platform
+  *)
+      case ${PW_CSP:-} in 
+         "aws" | "google" | "azure") MACHINE_ID=noaacloud ;;
+         *) PW_CSP="UNKNOWN"
+            MACHINE_ID=UNKNOWN ;;  # Unknown platform
+      esac 
```

Comment:
Hi @BrianCurtis-NOAA..it was done to consolidate all of the MACHINE_ID statements..Please let me know if we should revert. Thanks!

Reply:
Since there is a need to use this detect_machine.sh in global-workflow and its sub components this change isn't necessary enough to make. Please revert.

Reply:
@BrianCurtis-NOAA That makes sense. I have just pushed the revert. Thanks